src/OrchardCore.Modules/OrchardCore.Flows/Drivers/BagPartDisplay.cs

```diff
@@ -34,13 +34,15 @@ IServiceProvider serviceProvider
 
         public override IDisplayResult Display(BagPart bagPart, BuildPartDisplayContext context)
         {
-            return Shape<BagPartViewModel>("BagPart", m =>
+            var hasItems = bagPart.ContentItems.Any();
+
+            return Shape<BagPartViewModel>(hasItems ? "BagPart" : "BagPart_NoItems", m =>
             {
                 m.BagPart = bagPart;
                 m.BuildPartDisplayContext = context;
                 m.Settings = context.TypePartDefinition.Settings.ToObject<BagPartSettings>();
             })
-            .Location("Detail", "Content:5");
+            .Location("Detail", hasItems ? "Content:5" : "-");
```

Comment:
Conditional placement? But then you don't even show the Empty shape by default? So someone who want to update the empty shape also needs to change the placement? Why not just render it, it's empty.

Reply:
The idea was to treat empty flows and bags as if there was nothing to show by default (scenario outlined in #1305 explains why).

I've pushed up a change that removes the conditional placement, so if people want to hide zones if the Flow or Bag is empty, then they can still do so with placement.